From da67121ad60d938fce6151ce2a998e30b89e59dd Mon Sep 17 00:00:00 2001
From: Vincent Abriou <vincent.abriou@st.com>
Date: Wed, 19 Oct 2016 18:40:09 +0200
Subject: [PATCH 19/24] STM: waylandsink: fix bad no_wl_buffer statement

The "else" statement is at the wrong place.

Change-Id: I2eab61e54216c2c03a4c4be89966bf3521ce76df
Signed-off-by: Vincent Abriou <vincent.abriou@st.com>
Reviewed-on: https://gerrit.st.com/59270
Reviewed-by: Fabien DESSENNE <fabien.dessenne@st.com>
Tested-by: Fabien DESSENNE <fabien.dessenne@st.com>
---
 ext/wayland/gstwaylandsink.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 55fbe84..acd7c90 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -777,15 +777,15 @@ gst_wayland_sink_render (GstBaseSink * bsink, GstBuffer * buffer)
           goto no_wl_buffer_shm;
 
         gst_buffer_add_wl_buffer (to_render, wbuf, sink->display);
-      } else {
-        /* we cannot get a wl_buffer and if we do not have any pool to create
-         * one by ourself */
-        goto no_wl_buffer;
       }
 
       gst_buffer_map (buffer, &src, GST_MAP_READ);
       gst_buffer_fill (to_render, 0, src.data, src.size);
       gst_buffer_unmap (buffer, &src);
+    } else {
+      /* we cannot get a wl_buffer and if we do not have any pool to create
+       * one by ourself */
+      goto no_wl_buffer;
     }
   }
 
-- 
2.7.4

